### DISCUSSION TITLE
Ardupilot with Hardware in the Loop

### DISCUSSION THREAD

**Discussion by blakermchale** [microsoft/AirSim#3622] [Support Q&A] [answered]
I am currently running unreal engine and using ArduPilot on a Pixhawk 4. I saw there was support for setting up hardware in the loop (HITL) with PX4, but could not find anything on ArduPilot support besides the software in the loop (SITL). Has anyone configured Ardupilot for HITL with AirSIm? If so, how did you do that and what control ports did you set in the settings.json?

**Comment by bravepaulo** [comment]
Hi @blakermchale.
Have you tried to use the same parameters that are proposed in PX4 example?
I have worked with HITL using Airsim and PX4, maybe the settings on json file work out for ADP as well.

**Comment by blakermchale** [comment]
Hi @bravepaulo,
I tried that but it returns back `Error while receiving rotor control data - ErrorNo: -1`. One step that I cannot find an alternative for is the `selecting the HIL Quadrocopter X airframe` for pixhawk 4. Any ideas?

**Comment by rajat2004** [comment, marked as answer]
ArduPilot hasn't supported HITL since many years
